static/js/containers/HomePage_test.js

```diff
@@ -23,59 +28,304 @@ describe("HomePage", () => {
+    let renderComponent
+
+    beforeEach(() => {
+      renderComponent = helper.renderComponent.bind(helper)
+    })
+
+    const renderPage = async () => {
+      const [wrapper] = await renderComponent("/", [
+        actions.frontpage.get.requestType,
+        actions.frontpage.get.successType,
+        actions.subscribedChannels.get.requestType,
+        actions.subscribedChannels.get.successType,
+        SET_POST_DATA,
+        SET_CHANNEL_DATA
+      ])
+      return wrapper.update()
+    }
 
-  it("lists subscriptions", async () => {
-    const wrapper = await renderPage()
-    wrapper.find(SubscriptionsList).forEach(component => {
-      assert.deepEqual(component.props().subscribedChannels, channels)
+    it("should fetch frontpage, set post data, render", async () => {
+      const wrapper = await renderPage()
+      assert.deepEqual(wrapper.find(PostList).props().posts, postList)
     })
   })
 
-  it("should switch the sorting method when an option is selected", async () => {
-    const wrapper = await renderPage()
-
-    for (const sortType of VALID_POST_SORT_TYPES) {
-      await listenForActions(
-        [
-          actions.frontpage.get.requestType,
-          actions.frontpage.get.successType,
-          SET_POST_DATA
-        ],
-        () => {
-          const select = wrapper.find(".post-list-title").find("select")
-          select.simulate("change", { target: { value: sortType } })
+  describe("sorting", () => {
+    it("uses hot sorting by default", async () => {
+      const { inner } = await render()
+      assert.equal(inner.find("PostSortPicker").props().value, POSTS_SORT_HOT)
+    })
+
+    it("uses the query parameter to set the sort", async () => {
+      const { inner } = await render(
+        {},
+        {
+          location: {
+            search: "?sort=some_sort"
+          }
         }
       )
+      assert.equal(inner.find("PostSortPicker").props().value, "some_sort")
+    })
 
-      wrapper.update()
-      assert.equal(
-        wrapper.find(HomePage).props().location.search,
-        `?sort=${sortType}`
-      )
-    }
+    it("should switch the sorting method when an option is selected", async () => {
+      const { inner } = await render()
+      for (const sortType of VALID_POST_SORT_TYPES) {
+        inner
+          .find("PostSortPicker")
+          .props()
+          .updateSortParam({
+            target:         { value: sortType },
+            preventDefault: helper.sandbox.stub()
+          })
+
+        assert.equal(helper.currentLocation.search, `?sort=${sortType}`)
+      }
+    })
+  })
+  ;[
+    [true, true, true],
+    [true, false, false],
+    [false, true, true],
+    [false, false, true]
+  ].forEach(([processing, hasPostIds, expectedLoaded]) => {
+    it(`sets loaded to ${String(expectedLoaded)} when the API request is ${
+      processing ? "" : "not "
+    }in progress and there ${
+      hasPostIds ? "are" : "aren't"
+    } post ids`, async () => {
```

Comment:
The list of post ids can be defined outside of this function and passed in as a parameter. Also, the description could be easier to read

```javascript
;[
    [true, mockPostIds, true],
    [true, [], false],
    [false, mockPostIds, true],
    [false, [], true]
].forEach(([processing, postIds, expectedLoaded]) => {
  it(`sets loaded to ${String(expectedLoaded)} when there are ${String(postIds.length)} and API request processing=${String(processing)}
}
```

Reply:
I'm not sure that's possible since the posts are defined in `beforeEach` and the test name is evaluated before that

Reply:
That list (or a separate list) could be initialized outside of `beforeEach`. You could also include a small description as a parameter if that makes it more readable